metrics/sinks/influxdb/influxdb.go

```diff
@@ -198,16 +198,16 @@ func (sink *influxdbSink) createDatabase() error {
 	if sink.dbExists {
 		return nil
 	}
+
 	q := influxdb.Query{
-		Command: fmt.Sprintf(`CREATE DATABASE %s WITH NAME "default"`, sink.c.DbName),
+		Command: fmt.Sprintf(`CREATE DATABASE %s WITH DURATION %s REPLICATION 1 NAME "default"`, sink.c.DbName, sink.c.RetentionPolicy),
 	}
 
-	if resp, err := sink.client.Query(q); err != nil {
-		if !(resp != nil && resp.Err != nil && strings.Contains(resp.Err.Error(), "already exists")) {
-			err := sink.createRetentionPolicy()
-			if err != nil {
-				return err
-			}
+	if resp, err := sink.client.Query(q); err != nil || resp.Error() != nil {
+		if err != nil {
```

Comment:
sure but influxdb doesn't return an error if the db already exists so we are already fine.
Will find the doc reference to that fact (and I tested it, otherwise it would fail the second time it is launched)